src/main/java/seedu/address/model/person/MatricNumber.java

```diff
@@ -0,0 +1,54 @@
+package seedu.address.model.person;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.util.AppUtil.checkArgument;
+
+/**
+ * Represents a Person's matric number in the address book.
+ * Guarantees: immutable; is valid as declared in {@link #isValidMatricNumber(String)}
+ */
+public class MatricNumber {
+
+    public static final String MESSAGE_MATRIC_NUMBER_CONSTRAINTS =
+            "Person matric number must begin with one letter, digits in the middle and a letter at the end";
+
+    public static final String MATRIC_NUMBER_VALIDATION_REGEX = "^[a-zA-Z][\\d]+[a-zA-Z]$";
+
+    public final String value;
+
+    /**
+     * Constructs a {@code MatricNumber}.
+     *
+     * @param matricNumber A valid matric number.
+     */
+    public MatricNumber(String matricNumber) {
+        requireNonNull(matricNumber);
+        checkArgument(isValidMatricNumber(matricNumber), MESSAGE_MATRIC_NUMBER_CONSTRAINTS);
+        this.value = matricNumber;
+    }
+
+    /**
+     * Returns true if a given string is a valid person email.
+     */
+    public static boolean isValidMatricNumber(String test) {
+        return test.matches(MATRIC_NUMBER_VALIDATION_REGEX);
```

Comment:
is there a way to validate that there are exactly 7 digits between the letters, and that the first letter is an A?
Also, would it be better to validate the year of matriculation (can't be greater than the current year and can't be older than 6 years)? That would be more realistic as clubs can remove people based on their matric numbers as well.

Reply:
Made the changes in latest commit. Checking by matriculation year would be unrealistic since we dont know how NUS will issue the matriculation number in the future